### DISCUSSION TITLE
order by RAND() for mysql

### DISCUSSION THREAD

**Discussion by kylderiNascimento** [prisma/prisma#13846] [Q&A]
Hello guys, how do I implement a random orderBy in the query?
I use mySql database.

**Comment by austincrim** [comment]
Hey @kylderiNascimento 👋,

What are you trying to accomplish? Selecting a random row or something else?

You can select a random row in Prisma like so:
```ts
let random = await prisma.user.findMany({
  take: 1,
  skip: Math.floor(Math.random() * someMaximumNumber)
})
```

**Reply by kylderiNascimento** [reply]
Thanks @austincrim. I'm trying to load a list in an unordered way. We can sort a list either DESC or ASC. There would be some way to sort randomly, that is, the position of each line changes according to each request.

**Reply by austincrim** [reply]
I think your best bet is to use raw SQL with [`$queryRaw`](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw).

**Comment by stouch** [comment]
Why prisma does not allow `SELECT * FROM myTable ORDER BY RAND()` ?

Is this because of performance (https://jan.kneschke.de/projects/mysql/order-by-rand/) ?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Reply by magror** [reply]
Please prioritize this
